sql/core.go

```diff
@@ -429,22 +439,42 @@ type IndexAlterableTable interface {
 	RenameIndex(ctx *Context, fromIndexName string, toIndexName string) error
 }
 
-// ForeignKeyTable is a table that can declare its foreign key constraints.
+// ForeignKeyTable is a table that can declare its foreign key constraints, as well as be referenced.
 type ForeignKeyTable interface {
-	Table
-	// GetForeignKeys returns the foreign key constraints on this table.
-	GetForeignKeys(ctx *Context) ([]ForeignKeyConstraint, error)
-}
-
-// ForeignKeyAlterableTable represents a table that supports foreign key modification operations.
-type ForeignKeyAlterableTable interface {
-	Table
-	// CreateForeignKey creates an index for this table, using the provided parameters.
-	// Returns an error if the foreign key name already exists.
-	CreateForeignKey(ctx *Context, fkName string, columns []string, referencedTable string, referencedColumns []string,
-		onUpdate, onDelete ForeignKeyReferenceOption) error
-	// DropForeignKey removes a foreign key from the database.
+	IndexedTable
+	// CreateIndexForForeignKey creates an index for this table, using the provided parameters. Indexes created through
+	// this function are specifically ones generated for use with a foreign key. Returns an error if the index name
+	// already exists, or an index on the same columns already exists.
+	CreateIndexForForeignKey(ctx *Context, indexName string, using IndexUsing, constraint IndexConstraint, columns []IndexColumn) error
+
+	// GetDeclaredForeignKeys returns the foreign key constraints that are declared by this table.
+	GetDeclaredForeignKeys(ctx *Context) ([]ForeignKeyConstraint, error)
+	// GetReferencedForeignKeys returns the foreign key constraints that are referenced by this table.
+	GetReferencedForeignKeys(ctx *Context) ([]ForeignKeyConstraint, error)
+	// AddForeignKey adds the given foreign key constraint to the table. Returns an error if the foreign key name
+	// already exists on any other table within the database.
+	AddForeignKey(ctx *Context, fk ForeignKeyConstraint) error
+	// DropForeignKey removes a foreign key from the table.
 	DropForeignKey(ctx *Context, fkName string) error
+	// SetForeignKeyResolved is called once a foreign key has been resolved.
+	SetForeignKeyResolved(ctx *Context, fkName string) error
```

Comment:
what are the integrator's responsibilities when we tell them a FK has been resolved?

Reply:
To set the `IsResolved` flag to `true`, I'll update the comment to reflect that.